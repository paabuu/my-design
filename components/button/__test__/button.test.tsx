import 'jsdom-global/register';
import * as React from 'react';
import { render, mount } from 'enzyme';
import Button from '..';

describe('Button', () => {
    it('render correctly', () => {
        const wrapper = render(<Button>Follow</Button>)
        expect(wrapper).toMatchSnapshot();
        const wrapper1 = render(<Button type="primary">Primary</Button>);
        expect(wrapper1).toMatchSnapshot();
        const wrapper2 = render(<Button size="large">Large</Button>);
        expect(wrapper2).toMatchSnapshot();
    });

    it('render two zn chars correctly', () => {
        const wrapper = render(<Button>确定</Button>);
        expect(wrapper.text()).toBe('确 定');
    });

    it('simulate click event', async () => {
        const onClick = jest.fn();
        const wrapper = mount(<Button onClick={onClick}>Wave</Button>);
        expect(wrapper).toMatchSnapshot();
        wrapper.simulate('click');
        expect(onClick.mock.calls.length).toBe(1);
    });

    it('should combine classname', () => {
        const wrapper = mount(<Button className="new-class">ClassName</Button>);
        expect(wrapper).toMatchSnapshot();
    });
});