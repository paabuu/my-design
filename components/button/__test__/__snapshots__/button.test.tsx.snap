// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render correctly 1`] = `
<button
  class="md-btn"
>
  <span>
    Follow
  </span>
</button>
`;

exports[`Button render correctly 2`] = `
<button
  class="md-btn md-btn-primary"
>
  <span>
    Primary
  </span>
</button>
`;

exports[`Button render correctly 3`] = `
<button
  class="md-btn md-btn-large"
>
  <span>
    Large
  </span>
</button>
`;

exports[`Button should combine classname 1`] = `
<Button
  className="new-class"
>
  <button
    className="md-btn new-class"
    onClick={[Function]}
  >
    <span>
      ClassName
    </span>
  </button>
</Button>
`;

exports[`Button simulate click event 1`] = `
<Button
  onClick={[MockFunction]}
>
  <button
    className="md-btn"
    onClick={[Function]}
  >
    <span>
      Wave
    </span>
  </button>
</Button>
`;
